### PR TITLE
App/Service: Introduce NsdManager to register service on the network

### PR DESCRIPTION
**Pull request by niley7464**
This patch introduces NsdManager.
If NsdManager register service on the network,
client which connected to the same network could get IP address and port number.

### CODE REVIEW

File: ml_inference_offloading/src/main/java/ai/nnstreamer/ml/inference/offloading/MainService.kt [hunk 1]
```diff
@@ -83,6 +86,25 @@ class MainService : Service() {
     private lateinit var container: ServiceContainer
     private var initialized = false
     private var serverInfoMap = mutableMapOf<String, ServerInfo>()
+    private lateinit var nsdManager: NsdManager
+
+    private val registrationListener = object : NsdManager.RegistrationListener {
```
Comment (wooksong) [inline ml_inference_offloading/src/main/java/ai/nnstreamer/ml/inference/offloading/MainService.kt:91]: It looks OK, but is it necessary to create a sub-class of the object and use it as an argument?

Comment (niley7464) [reply ml_inference_offloading/src/main/java/ai/nnstreamer/ml/inference/offloading/MainService.kt]: Not necessary..
Also, it makes `listener already in use` error.

Comment (wooksong) [reply ml_inference_offloading/src/main/java/ai/nnstreamer/ml/inference/offloading/MainService.kt]: @niley7464 Does it need to be Singleton?

Comment (niley7464) [reply ml_inference_offloading/src/main/java/ai/nnstreamer/ml/inference/offloading/MainService.kt]: Do you mean `nsdManager`?

Comment (wooksong) [reply ml_inference_offloading/src/main/java/ai/nnstreamer/ml/inference/offloading/MainService.kt]: @niley7464 The instance of ```NsdManager.RegistrationListener```

Comment (niley7464) [reply ml_inference_offloading/src/main/java/ai/nnstreamer/ml/inference/offloading/MainService.kt]: I think listener should not be Singleton, so it is created everytime when registering service

File: ml_inference_offloading/src/main/java/ai/nnstreamer/ml/inference/offloading/MainService.kt [hunk 2]
```diff
@@ -120,6 +124,8 @@ class MainService : Service() {
             return
         }
 
+        nsdManager = (getSystemService(Context.NSD_SERVICE) as NsdManager)
```
Comment (wooksong) [inline ml_inference_offloading/src/main/java/ai/nnstreamer/ml/inference/offloading/MainService.kt:127]: Q: Is it possible to initialize ```nsdManager```  at line 89?

**Review by wooksong — APPROVED** [review]

**Pull request merged**